app/services/participant_add_service.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ParticipantAddService
+  attr_accessor :participant_profile
+
+  class << self
+    def call(early_career_teacher_profile)
+      new(early_career_teacher_profile).call
+    end
+  end
```

Comment:
Is this the same as
```
def self.call(early_career_teacher_profile)
  new(early_career_teacher_profile).call
end
```
? If so, could we stick with that syntax?

Reply:
It's not exactly the same. class << self is more flexible. For one it defines a block for the singleton class and allows methods within it to be scoped as public, protected or private. You have to just through hoops to do that with self. defined class instance methods.

Reply:
Right, and are we using any of this flexibility here?

I would rather have one consistent syntax for defining class methods, and we use `def self.method_name` in a bunch of places already. Plus, as I see it, this method is meant to be public, so using syntax that only lets you define public methods is fine. Or am I missing something and it is, in fact, private?

Reply:
I prefer the `class << self` style over the `self.xxx` style just on the basis that it makes the grouping of singleton object instance methods vs normal instance methods clearer and keeps them all together. That preference is just stylistic though. Personally I'd defer to MaK's superior knowledge of this stuff.

Reply:
I don't think this is big deal either way - we know clearly what both mean.